AdminDashboard/Main/Builders/AccountBuilder.cs

```diff
@@ -0,0 +1,23 @@
+﻿using AdminDashboard.Main.Factories;
+using AdminDashboard.Models.Domain;
+using AdminDashboard.Models.JsonRequests;
+
+namespace AdminDashboard.Main.Builders
+{
+    public class AccountBuilder : IAccountBuilder
```

Comment:
I have builders and factories throughout the application to abstract heavy instantiations!

Reply:
It looks that we have the following

```
AccountValidator
AccountBuilder
AccountOrchestrator
AccountRepository
Models/Domain/Account
```

and then we have `AccountController` to handle REST requests which uses all of the above.

To me it looks that actual business logic is spread out between multiple places (in an attempt to provide a clean separation of concerns) but at the end it fails to do so . We still do not have a clearn separation between service layers. For example, take a look at this HTTP handler.

```c#
[HttpPut("{accountId:guid}/User/{userId:guid}")]
        public async Task<IActionResult> RegisterUserToAccount(Guid accountId, Guid userId)
        {
            if (!await Validator.DoesAccountExist(accountId))
                return NotFound();

            if (await Validator.DoesAccountAlreadyHaveUser(accountId, userId))
                return BadRequest();

            if (await Validator.HasAccountReachedMaximumNumberOfUsers(accountId))
                return BadRequest(false);

            await Repository.AddUserAndAccountRelationship(accountId, userId);

            return Ok(true);
        }
```

This is clearly part of business logic but this code is handled by HTTP handler.